pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/Node.java

```diff
@@ -118,8 +180,18 @@
     // FIXME should not be inclusive
     int getEndColumn();
```

Comment:
It's related to this PR, but can be done in a second step: How about exposing the location of a node with a richer object than with 4 methods? So, instead of having `getBeginLine()`, `getBeginColumn()`, `getEndLine()` and `getEndColumn()`, we have a single method, e.g. `getPosition()` which returns a to-be-defined location/position object. I think, that would be a bit similar like what the Apex parser exposes.

Reply:
This would be nice yes ! I'd like to reuse the Document API for that, but its abstractions are a bit wonky for now, and I think it needs a cleanup before.